libraries/DataFlash/LogFile.cpp

```diff
@@ -1437,6 +1437,14 @@ void DataFlash_Class::Log_Write_Current_instance(const uint64_t time_us,
 // Write an Current data packet
 void DataFlash_Class::Log_Write_Current()
 {
+    // Big painful assert to ensure that logging won't produce suprising results when the
+    // number of battery monitors changes
+    static_assert(AP_BATT_MONITOR_MAX_INSTANCES == 9,
+                  "The number of batt monitors has changed without updating the log "
+                  "table entries. Please add new enums for LOG_CURRENT_MSG, LOG_CURRENT_CELLS_MSG "
+                  "directly following the highest indexed fields. Don't forget to update the log "
+                  "description table as well, Then adjust the hard coded value in the assert to match.");
+
     const uint64_t time_us = AP_HAL::micros64();
     const uint8_t num_instances = AP::battery().num_instances();
     if (num_instances >= 1) {
```

Comment:
Did you forget something? You added everything, but it's not logging the extra new instances.

Reply:
.... I did. And I can't stand this approach to logging. At a minimum this is moving to a for loop :/